p2p/client.go

```diff
@@ -69,26 +77,41 @@ type Client struct {
 	localPubsubServer *tmpubsub.Server
 
 	logger types.Logger
+
+	blocksync *BlockSync
+
+	blockSyncStore datastore.Datastore
+
+	latestSeenHeight uint64
+
+	// block already received and not necessary to request using block-sync
```

Comment:
how is this map getting filled?

Reply:
when a block is received from p2p is added to the map, to avoid requesting it again. i refactored the map access and protected with mutex